FAILED HostConnectionPoolSpec "without any connections establish the number of configured min-connections"

Multiple failures for different configurations on old and new pool impl. So, it's likely the test is broken

https://jenkins.akka.io:8498/view/Akka-http/job/akka-http-nightly-against-akka-master-on-2.12/352/testReport/akka.http.impl.engine.client/HostConnectionPoolSpec/NewPoolImplementation_on_AkkaHttpEngineTCP_should_without_any_connections_establish_the_number_of_configured_min_connections/

```
Error Message

assertion failed: timeout (3 seconds) during expectMsgClass waiting for class akka.http.impl.engine.client.HostConnectionPoolSpec$SetupWithServerProbes$2$ServerConnection

Stacktrace

java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsgClass waiting for class akka.http.impl.engine.client.HostConnectionPoolSpec$SetupWithServerProbes$2$ServerConnection
	at scala.Predef$.assert(Predef.scala:219)
	at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:508)
	at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:481)
	at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:481)
	at akka.testkit.TestKit.expectMsgType(TestKit.scala:882)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$SetupWithServerProbes$2.expectNextConnection(HostConnectionPoolSpec.scala:444)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anon$13.<init>(HostConnectionPoolSpec.scala:277)
	at akka.http.impl.engine.client.HostConnectionPoolSpec.$anonfun$testSet$19(HostConnectionPoolSpec.scala:275)
```

Also maybe related:

https://jenkins.akka.io:8498/job/pr-validator-akka-http/1821/testReport/junit/akka.http.impl.engine.client/HostConnectionPoolSpec/NewPoolImplementation_on_PassThrough_should_re_establish_min_connections_when_number_of_open_connections_falls_below_threshold/

```
Error Message

assertion failed: timeout (4500 milliseconds) during expectMsgClass waiting for class akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$SetupWithServerProbes$2$ServerConnection
Stacktrace

java.lang.AssertionError: assertion failed: timeout (4500 milliseconds) during expectMsgClass waiting for class akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$SetupWithServerProbes$2$ServerConnection
	at scala.Predef$.assert(Predef.scala:170)
	at akka.testkit.TestKitBase$class.expectMsgClass_internal(TestKit.scala:472)
	at akka.testkit.TestKitBase$class.expectMsgType(TestKit.scala:445)
	at akka.testkit.TestKit.expectMsgType(TestKit.scala:814)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$SetupWithServerProbes$2.expectNextConnection(HostConnectionPoolSpec.scala:444)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$$anonfun$apply$mcV$sp$19$$anon$14.<init>(HostConnectionPoolSpec.scala:287)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$$anonfun$apply$mcV$sp$19.apply(HostConnectionPoolSpec.scala:283)
	at akka.http.impl.engine.client.HostConnectionPoolSpec$$anonfun$testSet$1$$anonfun$apply$mcV$sp$19.apply(HostConnectionPoolSpec.scala:283)
```

--- a/akka-http-core/src/test/scala/akka/http/impl/engine/client/HostConnectionPoolSpec.scala
+++ b/akka-http-core/src/test/scala/akka/http/impl/engine/client/HostConnectionPoolSpec.scala
@@ -438,7 +438,7 @@ class HostConnectionPoolSpec extends AkkaSpec(
               .run()(singleElementBufferMaterializer)
         }
 
-        private val serverConnections = TestProbe()
+        private lazy val serverConnections = TestProbe()
 
         def expectNextConnection(): ServerConnection =
           serverConnections.expectMsgType[ServerConnection]
@@ -446,7 +446,7 @@ class HostConnectionPoolSpec extends AkkaSpec(
         def expectNoNewConnection(): Unit =
           serverConnections.expectNoMsg()
 
-        protected lazy val server =
+        protected override lazy val server =
           Flow.fromSinkAndSourceMat(
             // buffer is needed because the async subscriber/publisher boundary will otherwise request > 1
             Flow[HttpRequest].buffer(1, OverflowStrategy.backpressure)

